test/v1/controllers/requestParsers/validators/SubmitReturnValidatorSpec.scala

```diff
@@ -0,0 +1,285 @@
+      |  "vatDueAcquisitions": -100.45,
+      |  "totalVatDue": 5.05,
+      |  "vatReclaimedCurrPeriod": 105.15,
+      |  "netVatDue": 100.10,
+      |  "totalValueSalesExVAT": 300,
+      |  "totalValuePurchasesExVAT": 300,
+      |  "totalValueGoodsSuppliedExVAT": 3000,
+      |  "totalAcquisitionsExVAT": 3000,
+      |  "receivedAt": "2020-05-05T12:01:00Z",
+      |  "uniqueId": "0123456789123"
+      |}
+    """.stripMargin
+  )
+
+  private val invalidVatDueSalesRequestBodyJson: JsValue = Json.parse(
+    """
+      |{
+      |  "periodKey": "18A1",
+      |  "vatDueSales": 105.505,
+      |  "vatDueAcquisitions": -100.45,
+      |  "totalVatDue": 5.05,
+      |  "vatReclaimedCurrPeriod": 105.15,
+      |  "netVatDue": 100.10,
+      |  "totalValueSalesExVAT": 300,
+      |  "totalValuePurchasesExVAT": 300,
+      |  "totalValueGoodsSuppliedExVAT": 3000,
+      |  "totalAcquisitionsExVAT": 3000,
+      |  "receivedAt": "2020-05-05T12:01:00Z",
+      |  "uniqueId": "0123456789"
+      |}
+    """.stripMargin
+  )
+
+  private val allInvalidValueRequestBodyJson: JsValue = Json.parse(
+    """
+      |{
+      |  "periodKey": "period",
+      |  "vatDueSales": 105.505,
+      |  "vatDueAcquisitions": -100.453,
+      |  "totalVatDue": 5.052,
+      |  "vatReclaimedCurrPeriod": 105.154,
+      |  "netVatDue": 100.102,
+      |  "totalValueSalesExVAT": 300.99,
+      |  "totalValuePurchasesExVAT": 300.99,
+      |  "totalValueGoodsSuppliedExVAT": 3000.99,
+      |  "totalAcquisitionsExVAT": 3000.99,
+      |  "receivedAt": "time",
+      |  "uniqueId": "0123456789123"
+      |}
+          """.stripMargin
+  )
+
+  private val emptyRequestBodyJson: JsValue = Json.parse("""{}""")
+
+  private val nonValidRequestBodyJson: JsValue = Json.parse(
+    """
+      |{
+      |  "periodKey": "period",
+      |  "vatDueSales": 105.505,
+      |  "vatDueAcquisitions": -100.453,
+      |  "totalVatDue": 5.052,
+      |  "vatReclaimedCurrPeriod": 105.154,
+      |  "netVatDue": 100.102,
+      |  "totalValueSalesExVAT": 300.99,
+      |  "totalValueGoodsSuppliedExVAT": 3000.99,
+      |  "totalAcquisitionsExVAT": 3000.99,
+      |  "totalValuePurchasesExVAT": 300.99
+      |}
+    """.stripMargin
+  )
+
+  private val validRawRequestBody = AnyContentAsJson(validRequestBodyJson)
+  private val emptyRawRequestBody = AnyContentAsJson(emptyRequestBodyJson)
+  private val nonValidRawRequestBody = AnyContentAsJson(nonValidRequestBodyJson)
+  private val invalidPeriodKeyRequestBody = AnyContentAsJson(invalidPeriodKeyRequestBodyJson)
+  private val invalidReceivedAtRequestBody = AnyContentAsJson(invalidReceivedAtRequestBodyJson)
+  private val invalidUniqueIdRequestBody = AnyContentAsJson(invalidUniqueIdRequestBodyJson)
+  private val invalidVatDueSalesRequestBody = AnyContentAsJson(invalidVatDueSalesRequestBodyJson)
+  private val allInvalidValueRawRequestBody = AnyContentAsJson(allInvalidValueRequestBodyJson)
+
+  class Test extends MockAppConfig {
```

Comment:
I don't think you need all this. Are you using the App config?. I think you can just use this 
`val validator: SubmitReturnValidator = new SubmitReturnValidator()`